sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Metrics/ExecutableLinesWalker.cs

```diff
@@ -81,11 +92,48 @@ private void FindExecutableLines(SyntaxNode node)
 
                 case SyntaxKind.ArrayInitializerExpression:
                     executableLineNumbers.Add(node.GetLocation().GetLineNumberToReport());
-                    return;
+                    return true;
+
+                case SyntaxKind.StructDeclaration:
+                case SyntaxKind.ClassDeclaration:
+                    var typeDeclarationSyntax = (BaseTypeDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(typeDeclarationSyntax.AttributeLists);
+
+                case SyntaxKind.MethodDeclaration:
+                    var methodSyntax = (BaseMethodDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(methodSyntax.AttributeLists);
+
+                case SyntaxKind.PropertyDeclaration:
+                    var propertySyntax = (PropertyDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(propertySyntax.AttributeLists);
+
+                case SyntaxKind.EventDeclaration:
+                    var eventSyntax = (EventDeclarationSyntax)node;
+                    return !HasExcludedCodeAttribute(eventSyntax.AttributeLists);
 
                 default:
-                    return;
+                    return true;
             }
         }
+
+        private bool HasExcludedCodeAttribute(SyntaxList<AttributeListSyntax> attributeSyntaxLists)
+        {
+            var attributes = attributeSyntaxLists.SelectMany(attributeList => attributeList.Attributes);
+            return attributes
+                .OfType<AttributeSyntax>()

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to 'IEnumerable<AttributeSyntax>'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1905)